@types/react-wait not included in dependencies

**What is the current behavior?**

TypeScript build fails with error:
```
node_modules/react-use/lib/useWait.d.ts:1:33 - error TS7016: Could not find a declaration file for module 'react-wait'. 

'....../node_modules/react-wait/dist/react-wait.js' implicitly has an 'any' type.
  Try `npm install @types/react-wait` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-wait';`

1 import { useWait, Waiter } from 'react-wait';
                                  ~~~~~~~~~~~~
```

**Steps to reproduce it and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have extra dependencies other than `react-use`. Paste the link to your [JSFiddle](https://jsfiddle.net) or [CodeSandbox](https://codesandbox.io) example below:**

```
echo '{}' > package.json && \
  npm i typescript react-use@latest && \
  echo "import { useWait } from 'react-use'; console.log(useWait);" && \
  npx tsc --init && \
  npx tsc --noEmit
```

**What is the expected behavior?**

Contrary to the expected some TypeScript types should be included as `dependencies`. This is the case with `react-wait` on which typings for `node_modules/react-use/lib/useWait.d.ts` depend (so it should indeed be included as dependency, not devDependency). More on that can be found in [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#dependencies).

Simply reverting 49372ac should be enough to make things work again :smile:

**A little about versions:**  
- _OS_: macOS
- _Browser (vendor and version)_: n/a
- _React_: n/a
- _`react-use`_: 12.2.3
- _Did this worked in the previous package version?_ yes (before #644 was implemented) - 12.2.2 does not have the error

--- a/package.json
+++ b/package.json
@@ -45,6 +45,7 @@
   },
   "homepage": "https://github.com/streamich/react-use#readme",
   "dependencies": {
+    "@types/react-wait": "^0.3.0",
     "copy-to-clipboard": "^3.1.0",
     "nano-css": "^5.1.0",
     "react-fast-compare": "^2.0.4",
@@ -76,7 +77,6 @@
     "@testing-library/react-hooks": "2.0.3",
     "@types/jest": "24.0.18",
     "@types/react": "16.9.2",
-    "@types/react-wait": "0.3.0",
     "babel-core": "6.26.3",
     "babel-loader": "8.0.6",
     "babel-plugin-dynamic-import-node": "2.3.0",

